packages/react-form/src/hooks/form.ts

```diff
@@ -41,45 +53,71 @@ import {useSubmit} from './submit';
  *         />
  *         {title.error && <p className="error">{title.error}</p>}
  *       </div>
+ *       {paymentMethodFields.map((field, index) => {
+ *         <TextField onChange={field.cardNumber.onChange} value={field.cardNumber.value} />
+ *         <TextField onChange={field.cvv.onChange} value={field.cvv.value} />
+ *         <button type="button" onClick={() => removeItem(index)}>Remove Payment Method</button>
+ *       })}
  *       <button disabled={!dirty} onClick={reset}>Reset</button>
  *       <button type="submit" disabled={!dirty}>Submit</button>
+ *       <button type="button" onClick={addItem}>Add Payment Method</button>
  *     </form>
  *  );
  *```
  *
  * @param fields - A dictionary of `Field` objects, dictionaries of `Field` objects, and lists of dictionaries of `Field` objects. Generally, you'll want these to be generated by the other hooks in this package, either `useField` or `useList`. This will be returned back out as the `fields` property of the return value.
  *
  * @param onSubmit - An async function to handle submission of the form. If this function returns an object of `{status: 'fail', error: FormError[]}` then the submission is considered a failure. Otherwise, it should return an object with `{status: 'success'}` and the submission will be considered a success. `useForm` will also call all client-side validation methods for the fields passed to it. The `onSubmit` handler will not be called if client validations fails.
+ * @param dynamicLists - Pass in dynamic lists objects into the form
+ * @param makeCleanAfterSubmit
  * @returns An object representing the current state of the form, with imperative methods to reset, submit, validate, and clean. Generally, the returned properties correspond 1:1 with the specific hook/utility for their functionality.
  *
  * @remarks
  * **Building your own:** Internally, `useForm` is a convenience wrapper over `useDirty`, `useReset`, and `useSubmit`. If you only need some of its functionality, consider building a custom hook combining a subset of them.
  * **Subforms:** You can have multiple `useForm`s wrapping different subsets of a group of fields. Using this you can submit subsections of the form independently and have all the error and dirty tracking logic "just work" together.
  */
+
 export function useForm<T extends FieldBag>({
   fields,
   onSubmit,
+  dynamicLists,
   makeCleanAfterSubmit = false,
 }: {
   fields: T;
   onSubmit?: SubmitHandler<FormMapping<T, 'value'>>;
   makeCleanAfterSubmit?: boolean;
+  dynamicLists?: DynamicListBag;
 }): Form<T> {
-  const dirty = useDirty(fields);
-  const basicReset = useReset(fields);
+  const fieldsWithLists = useMemo(() => {
+    const fieldsWithList = {...fields};
+    if (dynamicLists) {
+      Object.entries(dynamicLists).forEach(([key, value]) => {
+        (fieldsWithList as any)[key] = value.fields;
+      });
+    }
+    return fieldsWithList;
```

Comment:
This might be a micro optimization but to avoid the spreading when there is no dynamic list we could do
```suggestion
    if (dynamicLists) {
      const fieldsWithList = {...fields};
      Object.entries(dynamicLists).forEach(([key, value]) => {
        (fieldsWithList as any)[key] = value.fields;
      });
      return fieldsWithList;
    }
    return fields;
```